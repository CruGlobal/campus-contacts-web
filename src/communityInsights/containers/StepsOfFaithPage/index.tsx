import React, { useContext } from 'react';
import { useTranslation } from 'react-i18next';
import moment from 'moment';

import Card from '../../components/Card';
import AppContext from '../../../appContext';
import StagesSummary from '../../components/StagesSummary';
import StepsChart from '../../components/StepsChart';
import ImpactInfo from '../../components/ImpactInfo';
import FiltersChart from '../../components/FiltersChart';

import {
    GET_IMPACT_REPORT_TAKEN,
    GET_TOTAL_STEPS_COMPLETED_REPORT,
    GET_STEPS_COMPLETED_REPORT,
    GET_STAGES_REPORT,
    GET_IMPACT_REPORT_REACHED,
    GET_STAGES_PEOPLE_REPORT,
} from './queries';

const StepsOfFaithPage = () => {
    const { t } = useTranslation('insights');
    const { orgId }: any = useContext(AppContext);

    return (
        <div>
            <ImpactInfo
                query={GET_IMPACT_REPORT_TAKEN}
                text={report =>
                    t('stepsOfFaith.taken', {
                        count:
                            report.community.impactReport
                                .othersStepsCompletedCount,
                        people:
                            report.community.impactReport
                                .othersStepsReceiversCompletedCount,
                    })
                }
                variables={{ id: orgId }}
            />
            <Card title={t('stepsOfFaith.totalCompleted')}>
                <StagesSummary
                    query={GET_TOTAL_STEPS_COMPLETED_REPORT}
                    variables={{ id: orgId }}
                    mapData={data =>
                        data.community.report.stagesReport.map(
                            (entry: any) => ({
                                stage: entry.stage.name,
                                icon: entry.stage.name
                                    .toLowerCase()
                                    .replace(' ', '-'),
                                count: entry.othersStepsCompletedCount,
                            }),
                        )
                    }
                />
            </Card>
            <Card
                title={t('stepsOfFaith.completed')}
                subtitle={t('stepsOfFaith.completedSubtitle')}
            >
                <FiltersChart
                    query={GET_STEPS_COMPLETED_REPORT}
                    variables={{ id: orgId }}
                    mapData={data =>
                        data.community.report.daysReport.map((row: any) => ({
                            ['total']: row.othersStepsCompletedCount,
                            ['stages']: row.stageResults.map((stage: any) => ({
                                name: stage.othersStepsCompletedCount,
                                count: stage.stage.name,
                            })),
                            ['date']: row.date,
                        }))
                    }
                    currentDate={moment().toDate()}
                    label={t('stepsOfFaith.legend')}
                />
            </Card>
            <Card
                title={t('stepsOfFaith.added')}
                subtitle={t('stepsOfFaith.addedSubtitle')}
            >
                <StepsChart
                    query={GET_STAGES_REPORT}
                    mapData={data =>
                        data.community.report.stagesReport.map((row: any) => ({
                            [t(
                                'stepsOfFaith.legendLabel',
                            )]: row.othersStepsAddedCount,
                            [t('stage')]: row.stage.name.toUpperCase(),
                        }))
                    }
                    label={t('stepsOfFaith.legendLabel')}
                    index={t('stage')}
                    variables={{
                        period: 'P10Y',
                        id: orgId,
                        endDate: moment().format(),
                    }}
                />
            </Card>
            <ImpactInfo
                query={GET_IMPACT_REPORT_REACHED}
                text={report =>
                    t('stepsOfFaith.reached', {
                        count:
                            report.community.impactReport
                                .membersStageProgressionCount,
                    })
                }
                variables={{ id: orgId }}
            />
            <Card
                title={t('stepsOfFaith.people')}
                subtitle={t('stepsOfFaith.peopleSubtitle')}
            >
                <StepsChart
                    query={GET_STAGES_PEOPLE_REPORT}
                    mapData={data =>
                        data.community.report.stagesReport.map((row: any) => ({
                            [t('stepsOfFaith.peopleLabel')]: row.contactCount,
                            [t('stage')]: row.stage.name.toUpperCase(),
                        }))
                    }
                    label={t('stepsOfFaith.peopleLabel')}
                    index={t('stage')}
                    variables={{
                        period: 'P10Y',
                        id: orgId,
                        endDate: moment().format(),
                    }}
                />
            </Card>
        </div>
    );
};

export default StepsOfFaithPage;
