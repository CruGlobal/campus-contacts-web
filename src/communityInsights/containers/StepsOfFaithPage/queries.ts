import gql from 'graphql-tag';

const GET_IMPACT_REPORT_TAKEN = gql`
    query impactReportOtherStepsCompleted($id: ID!) {
        community(id: $id) {
            impactReport {
                othersStepsCompletedCount
                othersStepsReceiversCompletedCount
            }
        }
    }
`;

const GET_TOTAL_STEPS_COMPLETED_REPORT = gql`
    query communityReportStagesOthersStepsCompleted(
        $period: String!
        $id: ID!
        $endDate: ISO8601DateTime!
    ) {
        community(id: $id) {
            report(period: $period, endDate: $endDate) {
                stagesReport {
                    nodes {
                        othersStepsCompletedCount
                        stage {
                            name
                        }
                    }
                }
            }
        }
    }
`;

const GET_STEPS_COMPLETED_REPORT = gql`
    query communityReportDaysOthersSteps(
        $period: String!
        $id: ID!
        $endDate: ISO8601DateTime!
    ) {
        community(id: $id) {
            report(period: $period, endDate: $endDate) {
                daysReport {
                    nodes {
                        date
                        othersStepsCompletedCount
                        stageResults {
                            nodes {
                                othersStepsCompletedCount
                                stage {
                                    name
                                }
                            }
                        }
                    }
                }
            }
        }
    }
`;

const GET_STAGES_REPORT = gql`
    query communityReportStagesOthersStepsAdded(
        $period: String!
        $id: ID!
        $endDate: ISO8601DateTime!
    ) {
        community(id: $id) {
            report(period: $period, endDate: $endDate) {
                stagesReport {
                    nodes {
                        othersStepsAddedCount
                        stage {
                            name
                        }
                    }
                }
            }
        }
    }
`;

const GET_IMPACT_REPORT_REACHED = gql`
    query impactReportMembersStageProgressionCount($id: ID!) {
        community(id: $id) {
            impactReport {
                membersStageProgressionCount
            }
        }
    }
`;

const GET_STAGES_PEOPLE_REPORT = gql`
    query communityReportStagesOthersContactCount(
        $period: String!
        $id: ID!
        $endDate: ISO8601DateTime!
    ) {
        community(id: $id) {
            report(period: $period, endDate: $endDate) {
                stagesReport {
                    nodes {
                        contactCount
                        stage {
                            name
                        }
                    }
                }
            }
        }
    }
`;

export {
    GET_IMPACT_REPORT_TAKEN,
    GET_IMPACT_REPORT_REACHED,
    GET_TOTAL_STEPS_COMPLETED_REPORT,
    GET_STEPS_COMPLETED_REPORT,
    GET_STAGES_REPORT,
    GET_STAGES_PEOPLE_REPORT,
};
