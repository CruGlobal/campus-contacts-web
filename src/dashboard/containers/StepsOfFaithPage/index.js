import React, { useContext } from 'react';
import { useTranslation } from 'react-i18next';
import moment from 'moment';
import _ from 'lodash';

import Card from '../../components/Card';
import AppContext from '../../appContext';
import StagesSummary from '../../components/StagesSummary';
import StepsChart from '../../components/StepsChart';
import ImpactInfo from '../../components/ImpactInfo';
import FiltersChart from '../../components/FiltersChart';

import {
    GET_IMPACT_REPORT_TAKEN,
    GET_TOTAL_STEPS_COMPLETED_REPORT,
    GET_STEPS_COMPLETED_REPORT,
    GET_STAGES_REPORT,
    GET_IMPACT_REPORT_REACHED,
    GET_STAGES_PEOPLE_REPORT,
} from './queries';

const StepsOfFaithPage = () => {
    const { t } = useTranslation('insights');
    const { orgId } = useContext(AppContext);

    return (
        <div>
            <ImpactInfo
                query={GET_IMPACT_REPORT_TAKEN}
                text={report =>
                    t('stepsOfFaith.taken', {
                        count: report.impactReport.stepsCount,
                        people: report.impactReport.receiversCount,
                    })
                }
                variables={{ organizationId: orgId }}
            />
            <Card title={t('stepsOfFaith.totalCompleted')}>
                <StagesSummary
                    query={GET_TOTAL_STEPS_COMPLETED_REPORT}
                    variables={{
                        organizationId: orgId,
                    }}
                    mapData={data =>
                        data.communityReport.communityStagesReport.map(
                            entry => ({
                                stage: entry.pathwayStage,
                                icon: entry.pathwayStage
                                    .toLowerCase()
                                    .replace(' ', '-'),
                                count: entry.otherStepsCompletedCount,
                            }),
                        )
                    }
                />
            </Card>
            <Card
                title={t('stepsOfFaith.completed')}
                subtitle={t('stepsOfFaith.completedSubtitle')}
            >
                <FiltersChart
                    query={GET_STEPS_COMPLETED_REPORT}
                    mapData={data =>
                        data.communityReport.dayReport.map(row => ({
                            ['total']: row.stepsWithOthersStepsCompletedCount,
                            ['stages']: row.communityStagesReport.map(
                                stage => ({
                                    name: stage.pathwayStage,
                                    count:
                                        stage.stepsWithOthersStepsCompletedCount,
                                }),
                            ),
                            ['date']: row.date.substring(0, 2),
                        }))
                    }
                    countAverage={data =>
                        Math.floor(_.sumBy(data, 'total') / data.length)
                    }
                    label={t('stepsOfFaith.legend')}
                />
            </Card>
            <Card
                title={t('stepsOfFaith.added')}
                subtitle={t('stepsOfFaith.addedSubtitle')}
            >
                <StepsChart
                    query={GET_STAGES_REPORT}
                    mapData={data =>
                        data.organizationPathwayStagesReport.map(row => ({
                            [t(
                                'stepsOfFaith.legendLabel',
                            )]: row.othersStepsAddedCount,
                            [t('stage')]: row.pathwayStage.name.toUpperCase(),
                        }))
                    }
                    label={t('stepsOfFaith.legendLabel')}
                    index={t('stage')}
                    variables={{
                        period: '',
                        organizationId: orgId,
                        endDate: moment().format(),
                    }}
                />
            </Card>
            <ImpactInfo
                query={GET_IMPACT_REPORT_REACHED}
                text={report =>
                    t('stepsOfFaith.reached', {
                        count: report.impactReport.stepsCount,
                    })
                }
                variables={{ organizationId: orgId }}
            />
            <Card
                title={t('stepsOfFaith.people')}
                subtitle={t('stepsOfFaith.peopleSubtitle')}
            >
                <StepsChart
                    query={GET_STAGES_PEOPLE_REPORT}
                    mapData={data =>
                        data.organizationPathwayStagesReport.map(row => ({
                            [t('stepsOfFaith.peopleLabel')]: row.memberCount,
                            [t('stage')]: row.pathwayStage.name.toUpperCase(),
                        }))
                    }
                    label={t('stepsOfFaith.peopleLabel')}
                    index={t('stage')}
                    variables={{
                        period: '',
                        organizationId: orgId,
                        endDate: moment().format(),
                    }}
                />
            </Card>
        </div>
    );
};

export default StepsOfFaithPage;

StepsOfFaithPage.propTypes = {};
