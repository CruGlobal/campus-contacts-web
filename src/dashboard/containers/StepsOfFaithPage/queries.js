import gql from 'graphql-tag';

const GET_IMPACT_REPORT_TAKEN = gql`
    query impactReport($organizationId: ID!) {
        impactReport(organizationId: $organizationId) {
            stepsCount
            receiversCount
        }
    }
`;

const GET_TOTAL_STEPS_COMPLETED_REPORT = gql`
    query communityReport {
        communityReport {
            communityStagesReport {
                pathwayStage
                otherStepsCompletedCount
            }
        }
    }
`;

const GET_STEPS_COMPLETED_REPORT = gql`
    query communityReport {
        communityReport {
            dayReport {
                date
                stepsWithOthersStepsCompletedCount
                communityStagesReport {
                    pathwayStage
                    stepsWithOthersStepsCompletedCount
                }
            }
        }
    }
`;

const GET_STAGES_REPORT = gql`
    query organizationStagesReport(
        $period: String!
        $organizationId: ID!
        $endDate: ISO8601DateTime!
    ) {
        organizationStagesReport(
            period: $period
            organizationId: $organizationId
            endDate: $endDate
        ) {
            memberCount
            stage {
                name
            }
            othersStepsAddedCount
        }
    }
`;

const GET_IMPACT_REPORT_REACHED = gql`
    query impactReport($organizationId: ID!) {
        impactReport(organizationId: $organizationId) {
            stepsCount
        }
    }
`;

const GET_STAGES_PEOPLE_REPORT = gql`
    query organizationStagesReport(
        $period: String!
        $organizationId: ID!
        $endDate: ISO8601DateTime!
    ) {
        organizationStagesReport(
            period: $period
            organizationId: $organizationId
            endDate: $endDate
        ) {
            memberCount
            stage {
                name
            }
        }
    }
`;

export {
    GET_IMPACT_REPORT_TAKEN,
    GET_IMPACT_REPORT_REACHED,
    GET_TOTAL_STEPS_COMPLETED_REPORT,
    GET_STEPS_COMPLETED_REPORT,
    GET_STAGES_REPORT,
    GET_STAGES_PEOPLE_REPORT,
};
