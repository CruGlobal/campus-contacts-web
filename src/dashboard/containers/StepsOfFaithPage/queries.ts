import gql from 'graphql-tag';

const GET_IMPACT_REPORT_TAKEN = gql`
    query impactReport($id: ID!) {
        community(id: $id) {
            impactReport {
                othersStepsCompletedCount
                othersStepsReceiversCompletedCount
            }
        }
    }
`;

const GET_TOTAL_STEPS_COMPLETED_REPORT = gql`
    query communityStagesReport(
        $period: String!
        $id: ID!
        $endDate: ISO8601DateTime!
    ) {
        community(id: $id) {
            report(period: $period, endDate: $endDate) {
                stagesReport {
                    othersStepsCompletedCount
                    stage {
                        name
                    }
                }
            }
        }
    }
`;

const GET_STEPS_COMPLETED_REPORT = gql`
    query communityDayReport(
        $period: String!
        $id: ID!
        $endDate: ISO8601DateTime!
    ) {
        community(id: $id) {
            report(period: $period, endDate: $endDate) {
                daysReport {
                    date
                    othersStepsCompletedCount
                    stageResults {
                        othersStepsCompletedCount
                        stage {
                            name
                        }
                    }
                }
            }
        }
    }
`;

const GET_STAGES_REPORT = gql`
    query communityStagesReport(
        $period: String!
        $id: ID!
        $endDate: ISO8601DateTime!
    ) {
        community(id: $id) {
            report(period: $period, endDate: $endDate) {
                stagesReport {
                    othersStepsAddedCount
                    stage {
                        name
                    }
                }
            }
        }
    }
`;

const GET_IMPACT_REPORT_REACHED = gql`
    query impactReport($id: ID!) {
        community(id: $id) {
            impactReport {
                membersStageProgressionCount
            }
        }
    }
`;

const GET_STAGES_PEOPLE_REPORT = gql`
    query communityStagesReport(
        $period: String!
        $id: ID!
        $endDate: ISO8601DateTime!
    ) {
        community(id: $id) {
            report(period: $period, endDate: $endDate) {
                stagesReport {
                    contactCount
                    stage {
                        name
                    }
                }
            }
        }
    }
`;

export {
    GET_IMPACT_REPORT_TAKEN,
    GET_IMPACT_REPORT_REACHED,
    GET_TOTAL_STEPS_COMPLETED_REPORT,
    GET_STEPS_COMPLETED_REPORT,
    GET_STAGES_REPORT,
    GET_STAGES_PEOPLE_REPORT,
};
