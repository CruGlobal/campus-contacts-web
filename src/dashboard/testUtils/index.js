import React, { ReactElement } from 'react';
import { ApolloProvider } from 'react-apollo-hooks';
import { render } from '@testing-library/react';

import { createApolloMockClient } from './apolloMockClient';
import defaultTheme from '../defaultTheme';
import { ThemeProvider } from 'emotion-theming';
import AppContext from '../appContext';

export const renderWithContext = (component, { mocks } = {}, context = {}) => {
    const mockApolloClient = createApolloMockClient(mocks);

    // Warning: don't call any functions in here that return new instances on every call.
    // All the props need to stay the same otherwise renderer won't work.
    const wrapper = ({ children }) => (
        <ApolloProvider client={mockApolloClient}>
            <AppContext.Provider value={context}>
                <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
            </AppContext.Provider>
        </ApolloProvider>
    );

    const renderResult = render(component, { wrapper });

    return {
        ...renderResult,
        snapshot: () => {
            expect(renderResult.container).toMatchSnapshot();
        },
    };
};
