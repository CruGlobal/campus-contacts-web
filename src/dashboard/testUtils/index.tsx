import React, { ReactNode } from 'react';
import { ApolloProvider } from 'react-apollo-hooks';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'emotion-theming';

import defaultTheme from '../defaultTheme';
import AppContext from '../appContext';

import { createApolloMockClient } from './apolloMockClient';

export const renderWithContext = (
    component: any,
    { mocks: mocks = {}, appContext = {} } = {},
) => {
    const mockApolloClient = createApolloMockClient(mocks);

    // Warning: don't call any functions in here that return new instances on every call.
    // All the props need to stay the same otherwise renderer won't work.
    const wrapper = ({ children }: { children: ReactNode }) => (
        <ApolloProvider client={mockApolloClient}>
            <AppContext.Provider value={appContext}>
                <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
            </AppContext.Provider>
        </ApolloProvider>
    );

    // @ts-ignore
    const renderResult = render(component, { wrapper });
    return {
        ...renderResult,
        snapshot: () => {
            expect(renderResult.container).toMatchSnapshot();
        },
    };
};
