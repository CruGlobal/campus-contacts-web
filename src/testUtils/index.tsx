import React, { ReactNode, ReactElement } from 'react';
import { ApolloProvider } from 'react-apollo-hooks';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'emotion-theming';
import { IMocks } from 'graphql-tools';

import defaultTheme from '../defaultTheme';
import {
    AppContext,
    AppContextValue,
    appContextDefaultValue,
} from '../appContext';

import { createApolloMockClient } from './apolloMockClient';

interface RenderWithContextParams {
    mocks?: IMocks;
    appContext?: AppContextValue;
}

export const renderWithContext = (
    component: ReactElement,
    {
        mocks: mocks = {},
        appContext = appContextDefaultValue,
    }: RenderWithContextParams = {},
) => {
    const mockApolloClient = createApolloMockClient(mocks);

    // Warning: don't call any functions in here that return new instances on every call.
    // All the props need to stay the same otherwise renderer won't work.
    const wrapper = ({ children }: { children?: ReactNode }) => (
        <ApolloProvider client={mockApolloClient}>
            <AppContext.Provider value={appContext}>
                <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
            </AppContext.Provider>
        </ApolloProvider>
    );

    const renderResult = render(component, { wrapper });
    return {
        ...renderResult,
        snapshot: () => {
            expect(renderResult.container).toMatchSnapshot();
        },
    };
};
