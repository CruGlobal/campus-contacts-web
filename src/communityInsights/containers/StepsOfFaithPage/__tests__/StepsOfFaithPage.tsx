import React from 'react';

import { renderWithContext } from '../../../../testUtils';

import StepsOfFaithPage from '..';

describe('<StepsOfFaithPage />', () => {
    it('should render properly in loading state', async () => {
        const { snapshot, unmount } = renderWithContext(<StepsOfFaithPage />, {
            appContext: {
                orgId: 1,
            },
        });
        snapshot();
        unmount();
    });

    // TODO: Add a test with mocking data when GraphQL schema contains above query
});
