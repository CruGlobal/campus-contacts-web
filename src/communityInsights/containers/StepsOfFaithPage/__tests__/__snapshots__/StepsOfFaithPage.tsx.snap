// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StepsOfFaithPage /> should render properly in loading state 1`] = `
.emotion-0 {
  font-weight: 300;
  font-size: 30px;
  line-height: 38px;
  color: #505256;
  margin-bottom: 0px;
  margin-top: 48px;
  padding-left: 21px;
}

.emotion-5 {
  box-shadow: 0px 2px 12px rgba(0,0,0,0.2);
  background: #ffffff;
  border-radius: 8px;
  padding-left: 24px;
  padding-right: 24px;
  padding-top: 43px;
  margin-top: 36px;
}

.emotion-1 {
  font-weight: 300;
  font-size: 36px;
  line-height: 38px;
  color: #505256;
  margin-bottom: 12px;
  margin-top: 0px;
}

.emotion-2 {
  font-size: 14px;
  line-height: 20px;
  font-weight: normal;
  color: #505256;
  margin-top: 0;
  margin-bottom: 0;
}

.emotion-4 {
  padding: 25px;
  margin: 0;
}

.emotion-3 {
  width: 100%;
  height: 190px;
  background-color: #ffffff;
  -webkit-animation: fade 2s infinite linear both;
  animation: fade 2s infinite linear both;
  border-radius: 8px;
}

.emotion-8 {
  width: 100%;
  height: 400px;
  background-color: #ffffff;
  -webkit-animation: fade 2s infinite linear both;
  animation: fade 2s infinite linear both;
  border-radius: 8px;
}

<div>
  <div>
    <h1
      class="emotion-0"
    >
      <span>
        Gathering your stats...
      </span>
    </h1>
    <div
      class="emotion-5"
    >
      <h1
        class="emotion-1"
      >
        Total Steps of Faith Completed
      </h1>
      <h2
        class="emotion-2"
      />
      <div
        class="emotion-4"
      >
        <div
          class="emotion-3"
          width="190"
        />
      </div>
    </div>
    <div
      class="emotion-5"
    >
      <h1
        class="emotion-1"
      >
        Steps of Faith Completed
      </h1>
      <h2
        class="emotion-2"
      >
        Total steps of faith completed with others by members since  was created on .
      </h2>
      <div
        class="emotion-4"
      >
        <div
          class="emotion-8"
          width="400"
        />
      </div>
    </div>
    <div
      class="emotion-5"
    >
      <h1
        class="emotion-1"
      >
        Steps of Faith Added
      </h1>
      <h2
        class="emotion-2"
      >
        Current total number of steps of faith with others added by members.
      </h2>
      <div
        class="emotion-4"
      >
        <div
          class="emotion-8"
          width="400"
        />
      </div>
    </div>
    <h1
      class="emotion-0"
    >
      <span>
        Gathering your stats...
      </span>
    </h1>
    <div
      class="emotion-5"
    >
      <h1
        class="emotion-1"
      >
        People Stages
      </h1>
      <h2
        class="emotion-2"
      >
        Current total number of people added by members.
      </h2>
      <div
        class="emotion-4"
      >
        <div
          class="emotion-8"
          width="400"
        />
      </div>
    </div>
  </div>
</div>
`;
